src/streams.c

```diff
@@ -98,6 +98,67 @@ Int READ_COMMAND ( void )
     return 1;
 }
 
+Obj FuncREAD_ALL_COMMANDS( Obj self, Obj stream, Obj echo )
+{
+    ExecStatus status;
+    Int result_count, result_capacity;
+    Obj result, result_list;
+
+    // 16 Results should be enough for everyone
+    result_count = 1;
+    result_capacity = 16;
+    result_list = NEW_PLIST( T_PLIST, result_capacity );
+    SET_LEN_PLIST(result_list, result_count);
+
+    result = NEW_PLIST( T_PLIST, 2 );
+    SET_LEN_PLIST(result, 1);
+    SET_ELM_PLIST(result, 1, False );
+
+    SET_ELM_PLIST(result_list, 1, result);
+
+    /* try to open the file                                                */
+    if (!OpenInputStream(stream) ) {
+        return result_list;
```

Comment:
Looking at your test case, I am surprised to see that an empty input also yields `[[false]]`, I would have expected an empty list... Perhaps you have good reasons for all this behavior, but then it should be (at least briefly) explained in source comments.

Otherwise, I'd start the function roughly like this in order to achieve the behavior I suggested:
```C
  if (!OpenInputStream(stream) )
    return Fail;

  result_list = NEW_PLIST( T_PLIST, result_capacity );
  SET_LEN__PLIST(result_list, 0);

  TLS(Input)->echo = (echo == True);

   do {
    ...
   } while(...);
```